### PR TITLE
builtins: add object.union_n

### PR DESCRIPTION
**Pull request by anderseknert**
Fixes: #4012

Signed-off-by: Anders Eknert <anders@eknert.com>

### CODE REVIEW

File: topdown/object.go [hunk 1]
```diff
@@ -26,6 +26,25 @@ func builtinObjectUnion(_ BuiltinContext, operands []*ast.Term, iter func(*ast.T
 	return iter(ast.NewTerm(r))
 }
 
+func builtinObjectUnionN(_ BuiltinContext, operands []*ast.Term, iter func(*ast.Term) error) error {
+	arr, err := builtins.ArrayOperand(operands[0].Value, 1)
```
Comment (srenatus) [inline topdown/object.go:30]: We could allow sets of objects, too. What do you think?

**Review by srenatus — DISMISSED** [review]

File: topdown/object.go (hunk 1 above)
Comment (anderseknert) [reply topdown/object.go:30]: https://github.com/open-policy-agent/opa/issues/4012#issuecomment-970077620

😉

Comment (srenatus) [reply topdown/object.go:30]: 😂 ignore me.

**Comment by anderseknert** [comment]
We good to merge despite the build failure @srenatus ? Looks unrelated to me..

it would be good to see this in Wasm later though! I actually made an attempt but pretty hard without being able to test it easily 😅

**Comment by srenatus** [comment]
> We good to merge despite the build failure @srenatus ? Looks unrelated to me..

Is it? I'd expect a capabilities.json addition, and it's not there; that's what the failing check is about.

**Comment by anderseknert** [comment]
Oh, serves me right for checking on mobile I guess. Is that file edited manually? Somehow I thought it wasn’t.

**Comment by srenatus** [comment]
`make generate` should do it. On mobile myself, though 😅

**Comment by srenatus** [comment]
We surely could have some extra information here: https://github.com/open-policy-agent/opa/blob/54f79cee8f49236b4fc8260e334bb1cca8c7ad9a/build/check-working-copy.sh#L25-L30

**Review by srenatus — APPROVED** [review]

**Comment by anderseknert** [comment]
Thanks @srenatus 👍 Feel free to submit the corresponding Wasm function.. I actually took a stab at that myself the other day, and I even had _something_ which **might** have worked as intended.. the problem was I didn't know how to run that on my M1 😅 I hope that situation will be resolved soon!

**Pull request merged**